### DISCUSSION TITLE
Duplicate IPs in the same VRF

### DISCUSSION THREAD

**Discussion by michaelyand** [netbox-community/netbox#17910] [General]
Hi,

So ENFORCE_GLOBAL_UNIQUE works only for the GLOBAL table.
Would it be possible to add same mechanics for VRF?

I don't see a reason having duplicate IPs in the same VRF either.

Thanks

**Comment by candlerb** [comment]
> So ENFORCE_GLOBAL_UNIQUE works only for the GLOBAL table.
> Would it be possible to add same mechanics for VRF?

It's already settable per VRF:

![image](https://github.com/user-attachments/assets/3389de9b-5906-4254-a678-265816fe16a4)

If you mean you want to enforce uniqueness *between* the global table and any particular VRF, then no, Netbox can't do that, except perhaps by use of Custom Validators.

Netbox's "VRF" model is a mixture of two concepts: both a VRF (as in routing table) and a domain of IP address uniqueness ("network namespace"). Since every "VRF" in Netbox is also a separate network namespace, by definition an IP address in any VRF never duplicates any IP address in any other VRF (or the global table).

It has been proposed in the past to have a separate network namespace model, but it was declined.

**Comment by michaelyand** [comment]
Wow, how did I miss that? :(
Sorry and thanks!
Thought it is also in config.py
